test/cast/all_of_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule OpenApiSpex.CastAllOfTest do
+  use ExUnit.Case
+  alias OpenApiSpex.{Cast, Schema}
+  alias OpenApiSpex.Cast.{Error, AllOf}
+
+  defp cast(ctx), do: AllOf.cast(struct(Cast, ctx))
+
+  describe "cast/1" do
+    test "allOf" do
+      schema = %Schema{allOf: [%Schema{type: :integer}, %Schema{type: :string}]}
+      assert {:ok, 1} = cast(value: "1", schema: schema)
+    end
+
+    test "allOf, uncastable schema" do
+      schema = %Schema{allOf: [%Schema{type: :integer}, %Schema{type: :string}]}
+      assert {:error, [error]} = cast(value: [:whoops], schema: schema)
+
+      assert Error.message(error) ==
+               "Failed to cast [:whoops] as integer. Value must be castable using `allOf` schemas listed."
```

Comment:
Following up with the discussion about dumping the value into the error message, for this particular case, you could show the given unexpected data type (`:array`) instead of dumping the value.

Reply:
I'll get the instances where I'm dumping the variables removed and use the type instead here in a few.